diff --git a/bgpd/bgpd.c b/bgpd/bgpd.c
index 7479885..fbce4fb 100644
--- a/bgpd/bgpd.c
+++ b/bgpd/bgpd.c
@@ -4525,8 +4525,11 @@ bool peer_active(struct peer *peer)
 		return false;
 
 	if (peer->bfd_config) {
-		if (bfd_session_is_down(peer->bfd_config->session))
+		if (bfd_session_is_admin_down(peer->bfd_config->session))
 			return false;
+		else if (peer_established(peer->connection) &&
+			   bfd_session_is_down(peer->bfd_config->session))
+				return false;
 	}
 
 	if (peer->afc[AFI_IP][SAFI_UNICAST] || peer->afc[AFI_IP][SAFI_MULTICAST]
diff --git a/lib/bfd.c b/lib/bfd.c
index 4535fc1..9898922 100644
--- a/lib/bfd.c
+++ b/lib/bfd.c
@@ -1340,3 +1340,8 @@ bool bfd_session_is_down(const struct bfd_session_params *session)
 	return session->bss.state == BSS_DOWN ||
 	       session->bss.state == BSS_ADMIN_DOWN;
 }
+
+bool bfd_session_is_admin_down(const struct bfd_session_params *session)
+{
+	return session->bss.state == BSS_ADMIN_DOWN;
+}
diff --git a/lib/bfd.h b/lib/bfd.h
index 48929a9..29c960a 100644
--- a/lib/bfd.h
+++ b/lib/bfd.h
@@ -465,6 +465,7 @@ extern int bfd_nht_update(const struct prefix *match,
 			  const struct zapi_route *route);
 
 extern bool bfd_session_is_down(const struct bfd_session_params *session);
+extern bool bfd_session_is_admin_down(const struct bfd_session_params *session);
 
 #ifdef __cplusplus
 }
