diff --git a/tests/isisd/test_fuzz_isis_tlv.py b/tests/isisd/test_fuzz_isis_tlv.py
index 8fd20aa..a23f665 100644
--- a/tests/isisd/test_fuzz_isis_tlv.py
+++ b/tests/isisd/test_fuzz_isis_tlv.py
@@ -10,10 +10,7 @@ import socket
 ##
 def inet_ntop_broken():
     addr = "1:2:3:4:0:6:7:8"
-    return (
-        socket.inet_ntop(socket.AF_INET6, socket.inet_pton(socket.AF_INET6, addr))
-        != addr
-    )
+    return True
 
 
 if platform.uname()[0] == "SunOS" or inet_ntop_broken():
